### PR TITLE
Added solution

### PR DESCRIPTION
**Pull request by AndrewPolishchuc**

### CODE REVIEW

File: src/main/java/movie/storage/dao/impl/MovieDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = HibernateUtil.getSessionFactory().openSession();
+        try {
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
```
Comment (andromash) [inline src/main/java/movie/storage/dao/impl/MovieDaoImpl.java]: You don't need to set id manually, Hibernate does it automatically

File: src/main/resources/hibernate.cfg.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        <property name="connection.driver_class">com.mysql.cj.jdbc.Driver</property>
+        <property name="connection.username">root</property>
+        <property name="connection.password">dijbsjia3#</property>
+        <property name="show_sql">true</property>
+        <property name="hibernate.hbm2ddl.auto">create-drop</property>
+
+        <mapping class="movie.storage.model.Movie"/>
+    </session-factory>
+
+</hibernate-configuration>
```
Comment (andromash) [inline src/main/resources/hibernate.cfg.xml]: empty line needed :)

**Review by andromash — COMMENTED** [review]
Good job!

File: src/main/java/movie/storage/dao/impl/MovieDaoImpl.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/movie/storage/dao/impl/MovieDaoImpl.java]: agree

File: src/main/java/movie/storage/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            }
+        }
+    }
+
+    @Override
+    public List<Movie> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Movie> getAllMoviesQuery = session.createQuery("from Movie", Movie.class);
+            return getAllMoviesQuery.getResultList();
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/movie/storage/dao/impl/MovieDaoImpl.java:43]: let's add catch block here too

File: src/main/java/movie/storage/exception/IncorrectDataHandlingException.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package movie.storage.exception;
+
+public class IncorrectDataHandlingException extends RuntimeException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/movie/storage/exception/IncorrectDataHandlingException.java]: exception's cause its incorrect data or incorrect way of handling data?

File: src/main/java/movie/storage/dao/impl/MovieDaoImpl.java (hunk 1 above)
Comment (AndrewPolishchuc) [reply src/main/java/movie/storage/dao/impl/MovieDaoImpl.java]: Done

File: src/main/resources/hibernate.cfg.xml (hunk 2 above)
Comment (AndrewPolishchuc) [reply src/main/resources/hibernate.cfg.xml]: Done

File: src/main/java/movie/storage/dao/impl/MovieDaoImpl.java (hunk 3 above)
Comment (AndrewPolishchuc) [reply src/main/java/movie/storage/dao/impl/MovieDaoImpl.java:43]: Agree

File: src/main/java/movie/storage/exception/IncorrectDataHandlingException.java (hunk 4 above)
Comment (AndrewPolishchuc) [reply src/main/java/movie/storage/exception/IncorrectDataHandlingException.java]: Changed to IncorrectDataException

File: src/main/java/movie/storage/dao/impl/MovieDaoImpl.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/movie/storage/dao/impl/MovieDaoImpl.java]: has it?

File: src/main/java/movie/storage/util/HibernateUtil.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    private static SessionFactory sessionFactory = initSessionFactory();
+
+    private HibernateUtil() {
+    }
+
+    private static SessionFactory initSessionFactory() {
+        try {
+            return new Configuration().configure().buildSessionFactory();
+        } catch (Exception e) {
+            throw new IncorrectDataException("Failed to build SessionFactory", e);
```
Comment (dmytro-hryhoruk) [inline src/main/java/movie/storage/util/HibernateUtil.java]: let's leave just runtime exception here, its different type of exception than in Dao

Comment (AndrewPolishchuc) [reply src/main/java/movie/storage/util/HibernateUtil.java]: Done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**